<!DOCTYPE html>
<html lang='en'>
<head>
<title>mupdf-1.12.0-security-roundup.patch\trunk - svntogit/community.git - Git clone of the 'community' repository
</title>
<meta name='generator' content='cgit v0.12'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://git.archlinux.org/svntogit/community.git/atom/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf' type='application/atom+xml'/>
<link rel='vcs-git' href='https://git.archlinux.org/svntogit/community.git' title='svntogit/community.git Git repository'/>
<link rel='vcs-git' href='git://git.archlinux.org/svntogit/community.git' title='svntogit/community.git Git repository'/>
<link rel='vcs-git' href='ssh://git.archlinux.org/srv/git/svntogit/community.git' title='svntogit/community.git Git repository'/>
</head>
<body>
	<div id="archnavbar"><!-- Arch Linux global navigation bar -->
		<div id="archnavbarlogo">
			<p><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more"></a></p>
		</div>
		<div id="archnavbarmenu">
			<ul id="archnavbarlist">
				<li id="anb-home"><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
				<li id="anb-packages"><a href="http://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
				<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
				<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
				<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and follow bugs">Bugs</a></li>
				<li id="anb-sec"><a href="https://security.archlinux.org/" title="Security Tracker">Security</a></li>
				<li id="anb-aur"><a href="https://aur.archlinux.org/" title="Arch Linux User Repository">AUR</a></li>
				<li id="anb-download"><a href="http://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
			</ul>
		</div>
	</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'><a href='/'>index</a> : <a title='svntogit/community.git' href='/svntogit/community.git/'>svntogit/community.git</a></td></tr>
<tr><td class='sub'>Git clone of the 'community' repository
</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/svntogit/community.git/?h=packages/mupdf'>summary</a><a href='/svntogit/community.git/refs/?h=packages/mupdf'>refs</a><a href='/svntogit/community.git/log/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>log</a><a class='active' href='/svntogit/community.git/tree/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>tree</a><a href='/svntogit/community.git/commit/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>commit</a><a href='/svntogit/community.git/diff/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>diff</a><a href='/svntogit/community.git/stats/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>stats</a></td><td class='form'><form class='right' method='get' action='/svntogit/community.git/log/trunk/mupdf-1.12.0-security-roundup.patch'>
<input type='hidden' name='h' value='packages/mupdf'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/svntogit/community.git/tree/?h=packages/mupdf'>root</a>/<a href='/svntogit/community.git/tree/trunk?h=packages/mupdf'>trunk</a>/<a href='/svntogit/community.git/tree/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>mupdf-1.12.0-security-roundup.patch</a></div><div class='content'>blob: c4ad2ec5cb6a8047ee2fefebd1672706b38b039f (<a href='/svntogit/community.git/plain/trunk/mupdf-1.12.0-security-roundup.patch?h=packages/mupdf'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
<a id='n83' href='#n83'>83</a>
<a id='n84' href='#n84'>84</a>
<a id='n85' href='#n85'>85</a>
<a id='n86' href='#n86'>86</a>
<a id='n87' href='#n87'>87</a>
<a id='n88' href='#n88'>88</a>
<a id='n89' href='#n89'>89</a>
<a id='n90' href='#n90'>90</a>
<a id='n91' href='#n91'>91</a>
<a id='n92' href='#n92'>92</a>
<a id='n93' href='#n93'>93</a>
<a id='n94' href='#n94'>94</a>
<a id='n95' href='#n95'>95</a>
<a id='n96' href='#n96'>96</a>
<a id='n97' href='#n97'>97</a>
<a id='n98' href='#n98'>98</a>
<a id='n99' href='#n99'>99</a>
<a id='n100' href='#n100'>100</a>
<a id='n101' href='#n101'>101</a>
<a id='n102' href='#n102'>102</a>
<a id='n103' href='#n103'>103</a>
<a id='n104' href='#n104'>104</a>
<a id='n105' href='#n105'>105</a>
<a id='n106' href='#n106'>106</a>
<a id='n107' href='#n107'>107</a>
<a id='n108' href='#n108'>108</a>
<a id='n109' href='#n109'>109</a>
<a id='n110' href='#n110'>110</a>
<a id='n111' href='#n111'>111</a>
<a id='n112' href='#n112'>112</a>
<a id='n113' href='#n113'>113</a>
<a id='n114' href='#n114'>114</a>
<a id='n115' href='#n115'>115</a>
<a id='n116' href='#n116'>116</a>
<a id='n117' href='#n117'>117</a>
<a id='n118' href='#n118'>118</a>
<a id='n119' href='#n119'>119</a>
<a id='n120' href='#n120'>120</a>
<a id='n121' href='#n121'>121</a>
<a id='n122' href='#n122'>122</a>
<a id='n123' href='#n123'>123</a>
<a id='n124' href='#n124'>124</a>
<a id='n125' href='#n125'>125</a>
<a id='n126' href='#n126'>126</a>
<a id='n127' href='#n127'>127</a>
<a id='n128' href='#n128'>128</a>
<a id='n129' href='#n129'>129</a>
<a id='n130' href='#n130'>130</a>
<a id='n131' href='#n131'>131</a>
</pre></td>
<td class='lines'><pre><code>From 22339500c243e564eadf564b5ae2925e1caf44a9 Mon Sep 17 00:00:00 2001
From: Sebastian Rasmussen &lt;sebras&#64;gmail.com&gt;
Date: Sun, 21 Jan 2018 21:08:07 +0100
Subject: [PATCH] Bug 698889: Handle unterminated PDF arrays gracefully.

Thanks to oss-fuzz for reporting this.
<span class="hl kwb">---</span>
 source/pdf/pdf-parse.c | 3 +++
 1 file changed, 3 insertions(+)

From 2a1611030030e18010a0ab1d69eda0359eb5f585 Mon Sep 17 00:00:00 2001
From: Sebastian Rasmussen &lt;sebras&#64;gmail.com&gt;
Date: Wed, 13 Dec 2017 21:14:19 +0100
Subject: [PATCH] Validate that /Size in trailer is in range.

<span class="hl kwb">---</span>
 source/pdf/pdf-xref.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

From 5722ebc5823381ee57c525cbc0d4dc627009979d Mon Sep 17 00:00:00 2001
From: Tor Andersson &lt;tor.andersson&#64;artifex.com&gt;
Date: Fri, 1 Dec 2017 16:25:39 +0100
Subject: [PATCH] Fix 698787: avoid using &quot;system()&quot; to copy files.

<span class="hl kwb">---</span>
 platform/x11/x11_main.c | 35 ++++++++++++++++++++++++++++++-----
 1 file changed, 30 insertions(+), 5 deletions(-)

From d9bc8c6f7fb2e3ec7035bebaaee0edcf59287705 Mon Sep 17 00:00:00 2001
From: Sebastian Rasmussen &lt;sebras&#64;gmail.com&gt;
Date: Mon, 22 Jan 2018 17:56:20 +0100
Subject: [PATCH] Bug 698885: When parsing PDF version, make sure to initialize
 buffer.

Thanks to oss-fuzz for reporting this.
<span class="hl kwb">---</span>
 source/pdf/pdf-xref.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/pdf/pdf-parse.c b/source/pdf/pdf-parse.c
index ff741dc..7904ebd 100644
<span class="hl kwb">--- a/source/pdf/pdf-parse.c</span>
<span class="hl kwa">+++ b/source/pdf/pdf-parse.c</span>
&#64;&#64; -401,6 +401,9 &#64;&#64; pdf_parse_array(fz_context *ctx, pdf_document *doc, fz_stream *file, pdf_lexbuf
 
 			switch (tok)
 			{
<span class="hl kwa">+			case PDF_TOK_EOF:</span>
<span class="hl kwa">+				fz_throw(ctx, FZ_ERROR_SYNTAX, &quot;array not closed before end of file&quot;);</span>
<span class="hl kwa">+</span>
 			case PDF_TOK_CLOSE_ARRAY:
 				op = ary;
 				goto end;
diff --git a/source/pdf/pdf-xref.c b/source/pdf/pdf-xref.c
index f2ba5a5..392adf3 100644
<span class="hl kwb">--- a/source/pdf/pdf-xref.c</span>
<span class="hl kwa">+++ b/source/pdf/pdf-xref.c</span>
&#64;&#64; -748,8 +748,8 &#64;&#64; pdf_xref_size_from_old_trailer(fz_context *ctx, pdf_document *doc, pdf_lexbuf *b
 		trailer = pdf_parse_dict(ctx, doc, doc-&gt;file, buf);
 
 		size = pdf_to_int(ctx, pdf_dict_get(ctx, trailer, PDF_NAME_Size));
<span class="hl kwb">-		if (!size)</span>
<span class="hl kwb">-			fz_throw(ctx, FZ_ERROR_GENERIC, &quot;trailer missing Size entry&quot;);</span>
<span class="hl kwa">+		if (size &lt; 0 || size &gt; PDF_MAX_OBJECT_NUMBER + 1)</span>
<span class="hl kwa">+			fz_throw(ctx, FZ_ERROR_GENERIC, &quot;trailer Size entry out of range&quot;);</span>
 	}
 	fz_always(ctx)
 	{
diff --git a/platform/x11/x11_main.c b/platform/x11/x11_main.c
index fe2daa8..edbb9fa 100644
<span class="hl kwb">--- a/platform/x11/x11_main.c</span>
<span class="hl kwa">+++ b/platform/x11/x11_main.c</span>
&#64;&#64; -317,13 +317,38 &#64;&#64; void winreplacefile(char *source, char *target)
 
 void wincopyfile(char *source, char *target)
 {
<span class="hl kwb">-	char *buf = malloc(strlen(source)+strlen(target)+5);</span>
<span class="hl kwb">-	if (buf)</span>
<span class="hl kwa">+	FILE *in, *out;</span>
<span class="hl kwa">+	char buf[32 &lt;&lt; 10];</span>
<span class="hl kwa">+	int n;</span>
<span class="hl kwa">+</span>
<span class="hl kwa">+	in = fopen(source, &quot;rb&quot;);</span>
<span class="hl kwa">+	if (!in)</span>
<span class="hl kwa">+	{</span>
<span class="hl kwa">+		winerror(&amp;gapp, &quot;cannot open source file for copying&quot;);</span>
<span class="hl kwa">+		return;</span>
<span class="hl kwa">+	}</span>
<span class="hl kwa">+	out = fopen(target, &quot;wb&quot;);</span>
<span class="hl kwa">+	if (!out)</span>
<span class="hl kwa">+	{</span>
<span class="hl kwa">+		winerror(&amp;gapp, &quot;cannot open target file for copying&quot;);</span>
<span class="hl kwa">+		fclose(in);</span>
<span class="hl kwa">+		return;</span>
<span class="hl kwa">+	}</span>
<span class="hl kwa">+</span>
<span class="hl kwa">+	for (;;)</span>
 	{
<span class="hl kwb">-		sprintf(buf, &quot;cp %s %s&quot;, source, target);</span>
<span class="hl kwb">-		system(buf);</span>
<span class="hl kwb">-		free(buf);</span>
<span class="hl kwa">+		n = fread(buf, 1, sizeof buf, in);</span>
<span class="hl kwa">+		fwrite(buf, 1, n, out);</span>
<span class="hl kwa">+		if (n &lt; sizeof buf)</span>
<span class="hl kwa">+		{</span>
<span class="hl kwa">+			if (ferror(in))</span>
<span class="hl kwa">+				winerror(&amp;gapp, &quot;cannot read data from source file&quot;);</span>
<span class="hl kwa">+			break;</span>
<span class="hl kwa">+		}</span>
 	}
<span class="hl kwa">+</span>
<span class="hl kwa">+	fclose(out);</span>
<span class="hl kwa">+	fclose(in);</span>
 }
 
 void cleanup(pdfapp_t *app)
diff --git a/source/pdf/pdf-xref.c b/source/pdf/pdf-xref.c
index 392adf3..4997ebe 100644
<span class="hl kwb">--- a/source/pdf/pdf-xref.c</span>
<span class="hl kwa">+++ b/source/pdf/pdf-xref.c</span>
&#64;&#64; -590,7 +590,7 &#64;&#64; pdf_load_version(fz_context *ctx, pdf_document *doc)
 
 	fz_seek(ctx, doc-&gt;file, 0, SEEK_SET);
 	fz_read_line(ctx, doc-&gt;file, buf, sizeof buf);
<span class="hl kwb">-	if (memcmp(buf, &quot;%PDF-&quot;, 5) != 0)</span>
<span class="hl kwa">+	if (strlen(buf) &lt; 5 || memcmp(buf, &quot;%PDF-&quot;, 5) != 0)</span>
 		fz_throw(ctx, FZ_ERROR_GENERIC, &quot;cannot recognize version marker&quot;);
 
 	doc-&gt;version = 10 * (fz_atof(buf+5) + 0.05f);
<span class="hl kwb">-- </span>
2.9.1
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class="foot" style="padding-left:1em;padding-right:1em;">
<p>Copyright &copy; 2002-2017 <a href="mailto:jvinet@zeroflux.org"
title="contact Judd Vinet">Judd Vinet</a> and <a href="mailto:aaron@archlinux.org"
title="contact Aaron Griffin">Aaron Griffin</a>. The Arch Linux name and logo
are recognized trademarks. Some rights reserved. The registered trademark
Linux&reg; is used pursuant to a sublicense from LMI, the exclusive licensee
of Linus Torvalds, owner of the mark on a world-wide basis.</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
